### PR TITLE
Create System.Private.CoreLib assembly

### PR DESCRIPTION
**Pull request by AlexGhiondea**
Create System.Private.CoreLib assembly that contains the same code as mscorlib but with a different name.

/cc @gkhanna79 @jkotas @weshaggard

### CODE REVIEW

File: src/build.proj [hunk 1]
```diff
@@ -4,6 +4,7 @@
   <!-- List the projects that need to be built -->
   <ItemGroup>
     <Project Include="mscorlib\mscorlib.csproj" />
+    <Project Include="mscorlib\System.Private.CoreLib.csproj" />
```
Comment (gkhanna79) [inline src/build.proj]: I presume this file  (build.proj) is not used in the closed build, right?

**Comment by gkhanna79** [comment]
The change looks good to me. I have one suggestion - can you rename "src\mscorlib" folder to "src\S.P.C"?

File: src/build.proj (hunk 1 above)
Comment (AlexGhiondea) [inline src/build.proj]: Yes, I believe it is not used in the closed build.

**Comment by gkhanna79** [comment]
@AlexGhiondea Do you expect to merge this today?

File: src/mscorlib/Common/Preprocessed/AssemblyRefs.CoreLib.g.cs [hunk 2]
```diff
@@ -0,0 +1,1082 @@
...
+
+using System;
+using System.Reflection;
+using System.Resources;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+
+// #line 1 "d:\\projectk_2\\src\\ndp\\clr\\src\\bcl.oss\\open\\src\\common\\include\\fxver.h"
```
Comment (weshaggard) [inline src/mscorlib/Common/Preprocessed/AssemblyRefs.CoreLib.g.cs]: Do we really need this file checked in like this? I assume we already did it for mscorlib as well but it seems like it should be cleaned up to only contain what is necessary and not all the hardcoded string paths.

File: src/mscorlib/System.Private.CoreLib.csproj [hunk 3]
```diff
@@ -0,0 +1,198 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (weshaggard) [inline src/mscorlib/System.Private.CoreLib.csproj]: Is it safe to assume this file is identical to mscorlib.csproj?

**Comment by AlexGhiondea** [comment]
@gkhanna79 yes -- I will merge shortly.

File: src/mscorlib/Common/Preprocessed/AssemblyRefs.CoreLib.g.cs (hunk 2 above)
Comment (AlexGhiondea) [inline src/mscorlib/Common/Preprocessed/AssemblyRefs.CoreLib.g.cs]: We used to need this. We might not need to have it anymore though.

File: src/mscorlib/System.Private.CoreLib.csproj (hunk 3 above)
Comment (AlexGhiondea) [inline src/mscorlib/System.Private.CoreLib.csproj]: Yes, with the assembly file name changed and referencing the corelib file (for model.xml and the preprocessed file).

**Pull request merged**